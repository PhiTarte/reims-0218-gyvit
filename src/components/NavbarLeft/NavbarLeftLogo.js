import React from "react"
import { Col } from "reactstrap"
import { NavLink } from "react-router-dom"

const NavbarLeftLogo = () => {
  return (
    <div
      className="d-flex align-items-center"
      style={{
        color: "white",
        fontWeight: "bold",
        height: "90px",
        marginBottom: "0px",
        textAlign: "center",
        backgroundColor: "#181048",
        fontSize: "20px"
      }}
    >
      <Col>
        <NavLink style={{ color: "white" }} to="/Home">
          GYVIT
        </NavLink>
      </Col>
    </div>
  )
}

export default NavbarLeftLogo
