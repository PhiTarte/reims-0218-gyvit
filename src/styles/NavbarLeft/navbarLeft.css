.navbar-left {
  width: 100%;
  height: 100vh;
  background-color: #231b56;
  color: white; }
