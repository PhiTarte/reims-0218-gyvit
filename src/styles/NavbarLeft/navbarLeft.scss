@import "../variables.scss";

.navbar-left {
  width: 100%;
  height: 100vh;
  background-color: $primary-color;
  color: white;
}
