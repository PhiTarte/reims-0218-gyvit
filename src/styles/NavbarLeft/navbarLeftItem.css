.nav-item {
  margin-right: 0px;
  margin-left: 25%;
  margin-top: 30%;
  font-size: 16px;
  text-align: left; }
