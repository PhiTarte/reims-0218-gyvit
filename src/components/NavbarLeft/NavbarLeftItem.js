// TODO create navbar

import React from "react"
import PropTypes from "prop-types"
import { Row, Col } from "reactstrap"
import { NavLink } from "react-router-dom"
import "./styles-navbar-left/navbarLeftItem.css"

const style = {
  color: "#00BFFF",
 // textDecoration: "none"
}

const NavbarLeftItem = ({ icon, nameLink, link }) => (
  <div>
    <Row>
      <Col xs="12">
        <NavLink to={link} activeStyle={style} 
        style = {{
            textDecoration : "none",
        }} className="navbar-left-item">
          <img alt="" src={icon} className="navbar-left-icon" />
          {nameLink}
        </NavLink>
      </Col>
    </Row>
    <hr />
  </div>
)

NavbarLeftItem.propTypes = {
  icon: PropTypes.string.isRequired,
  nameLink: PropTypes.string.isRequired
}

export default NavbarLeftItem
