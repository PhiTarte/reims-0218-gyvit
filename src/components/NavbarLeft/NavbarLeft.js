// TODO list of links to navbar

import React from "react"
import NavbarLeftItem from "./NavbarLeftItem"
import { Nav, NavItem } from "reactstrap"
import NavbarLeftLogo from "./NavbarLeftLogo"

const NavbarLeft = ({ dataLinks }) => (
  <div>
    <NavbarLeftLogo />
    <Nav
      vertical
      light
      expand="md"
      style={{
        height: "1024px",
        backgroundColor: "#231b56",
        color: "white"
      }}
    >
      <NavItem>
        {dataLinks.map((link, index) => (
          <NavbarLeftItem key={index} {...link} />
        ))}
      </NavItem>
    </Nav>
  </div>
)

export default NavbarLeft
