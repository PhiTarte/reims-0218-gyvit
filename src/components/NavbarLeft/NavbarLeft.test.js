//TODO test navbar list of links

import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import NavbarLeft from "./NavbarLeft"

const listLinksProps = {
  dataLinks: [
    {
      nameLink: "Accueil",
      icon: process.env.PUBLIC_URL + "/img/house_38533.ico"
    },
    {
      nameLink: "Fichiers",
      icon: process.env.PUBLIC_URL + "/img/files.ico"
    },
    {
      nameLink: "Partages",
      icon: process.env.PUBLIC_URL + "/img/share.ico"
    },
    {
      nameLink: "Modèles",
      icon: process.env.PUBLIC_URL + "/img/templates.ico"
    },
    {
      nameLink: "Demandes",
      icon: process.env.PUBLIC_URL + "/img/request.ico"
    },
    {
      nameLink: "Liens public",
      icon: process.env.PUBLIC_URL + "/img/links.ico"
    },
    {
      nameLink: "Contacts",
      icon: process.env.PUBLIC_URL + "/img/contact.ico"
    },
    {
      nameLink: "Paramètres",
      icon: process.env.PUBLIC_URL + "/img/settings.ico"
    }
  ]
}

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<NavbarLeft {...listLinksProps} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders link html", () => {
  const tree = renderer.create(<NavbarLeft {...listLinksProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})
