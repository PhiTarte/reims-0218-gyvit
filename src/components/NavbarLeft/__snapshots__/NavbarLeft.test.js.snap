// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link html 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <ul
        className="nav flex-column"
        expand="md"
        style={
          Object {
            "backgroundColor": "#231b56",
            "color": "white",
            "height": "1024px",
            "width": "220px",
          }
        }
      >
        <a
          className="navbar-brand"
          href="/"
          style={
            Object {
              "backgroundColor": "#181048",
              "color": "white",
              "fontWeight": "bold",
              "height": "90px",
              "textAlign": "center",
              "width": "220px",
            }
          }
        >
          <p>
            GYVIT
          </p>
        </a>
        <li
          className="text-center nav-item"
        >
          <div>
            <div
              className="container"
            >
              <div
                className="row"
              >
                <a
                  className="nav-link"
                  href="#"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "opacity": 0.5,
                    }
                  }
                >
                  <img
                    alt=""
                    src="/img/house_38533.ico"
                    style={
                      Object {
                        "opacity": 0.8,
                        "paddingRight": 10,
                      }
                    }
                  />
                  Home
                </a>
              </div>
            </div>
            <hr />
          </div>
          <div>
            <div
              className="container"
            >
              <div
                className="row"
              >
                <a
                  className="nav-link"
                  href="#"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "opacity": 0.5,
                    }
                  }
                >
                  <img
                    alt=""
                    src="/img/files.ico"
                    style={
                      Object {
                        "opacity": 0.8,
                        "paddingRight": 10,
                      }
                    }
                  />
                  Files
                </a>
              </div>
            </div>
            <hr />
          </div>
          <div>
            <div
              className="container"
            >
              <div
                className="row"
              >
                <a
                  className="nav-link"
                  href="#"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "opacity": 0.5,
                    }
                  }
                >
                  <img
                    alt=""
                    src="/img/share.ico"
                    style={
                      Object {
                        "opacity": 0.8,
                        "paddingRight": 10,
                      }
                    }
                  />
                  Shares
                </a>
              </div>
            </div>
            <hr />
          </div>
          <div>
            <div
              className="container"
            >
              <div
                className="row"
              >
                <a
                  className="nav-link"
                  href="#"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "opacity": 0.5,
                    }
                  }
                >
                  <img
                    alt=""
                    src="/img/templates.ico"
                    style={
                      Object {
                        "opacity": 0.8,
                        "paddingRight": 10,
                      }
                    }
                  />
                  Templates
                </a>
              </div>
            </div>
            <hr />
          </div>
          <div>
            <div
              className="container"
            >
              <div
                className="row"
              >
                <a
                  className="nav-link"
                  href="#"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "opacity": 0.5,
                    }
                  }
                >
                  <img
                    alt=""
                    src="/img/request.ico"
                    style={
                      Object {
                        "opacity": 0.8,
                        "paddingRight": 10,
                      }
                    }
                  />
                  Requests
                </a>
              </div>
            </div>
            <hr />
          </div>
          <div>
            <div
              className="container"
            >
              <div
                className="row"
              >
                <a
                  className="nav-link"
                  href="#"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "opacity": 0.5,
                    }
                  }
                >
                  <img
                    alt=""
                    src="/img/links.ico"
                    style={
                      Object {
                        "opacity": 0.8,
                        "paddingRight": 10,
                      }
                    }
                  />
                  Public links
                </a>
              </div>
            </div>
            <hr />
          </div>
          <div>
            <div
              className="container"
            >
              <div
                className="row"
              >
                <a
                  className="nav-link"
                  href="#"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "opacity": 0.5,
                    }
                  }
                >
                  <img
                    alt=""
                    src="/img/contact.ico"
                    style={
                      Object {
                        "opacity": 0.8,
                        "paddingRight": 10,
                      }
                    }
                  />
                  Contacts
                </a>
              </div>
            </div>
            <hr />
          </div>
          <div>
            <div
              className="container"
            >
              <div
                className="row"
              >
                <a
                  className="nav-link"
                  href="#"
                  onClick={[Function]}
                  style={
                    Object {
                      "color": "white",
                      "opacity": 0.5,
                    }
                  }
                >
                  <img
                    alt=""
                    src="/img/settings.ico"
                    style={
                      Object {
                        "opacity": 0.8,
                        "paddingRight": 10,
                      }
                    }
                  />
                  Settings
                </a>
              </div>
            </div>
            <hr />
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
